client/ui/login.go

```diff
@@ -2,53 +2,122 @@
-				qmlStatus.SendStatus("Loging failed")
-				return
-			}
-			//there we need to send to server loging data
-			//
-			loger.Log.Info("Connected to server")
-			qmlStatus.SendStatus("Login successfull")
-			qmlLogin.LoginDataIsValid(true)
-		} else {
-			qmlLogin.LoginDataIsValid(false)
+	qmlLogin.ConnectCheckLoginData(checkLoginDataAndConnect)
+	qmlLogin.ConnectLogOut(logOut)
+}
+
+func checkLoginDataAndConnect(userName, password string) {
+	if userName == "" || password == "" {
+		/*loger.Log.Info("Login failed. userName and/or password field are empty")
+		qmlStatus.SendStatus("Please, fill the fields")
+		qmlLogin.LoginDataIsValid(false)
+		return*/
+		qmlStatus.SendStatus("Test login.")
+		loger.Log.Info("Test login.")
+		qmlLogin.LoginDataIsValid(true)
+		userinfo.CurrentUserInfo = &userinfo.UserInfo{
+			UserName: "Kitler",
 		}
-	})
-	qmlLogin.ConnectLogOut(func(){
-		loger.Log.Info("Connection closed")
-		connection.Close()
-	})
-}
\ No newline at end of file
+		return
+	}
+	//err has separate declarations
+	// because in next statement we need to use global "connection" variable
+	//instead declarate new one
+	var err error
+	//another separate declaration/definition for connUrl
+	//because compiler emit error for declaration->definition->method_call in one statement
+	connUrl := url.URL{
+		Scheme: "ws",
+		Host:   config.GetConfig().Server.Host,
+		Path:   config.GetConfig().Server.Path,
+	}
+	connection, _, err = websocket.DefaultDialer.Dial(connUrl.String(), nil)
+	if err != nil {
+		loger.Log.Errorf("Can not dial. %s", err)
+		qmlLogin.LoginDataIsValid(false)
+		qmlStatus.SendStatus("Loging failed")
+		return
+	}
+	//there we need to send to server loging data
+	newMessageHeader := messageService.MessageHeader{
+		Type_:    config.GetConfig().MessageType.Auth,
+		Command:  config.GetConfig().MessageCommand.SendLoginData,
+		UserName: "",
+		Token:    "",
+	}
+	newMessageBody := messageService.User{
+		UserName: userName,
+		NickName: "",
+		Password: password,
+	}
+	newRawMessageBody, err := json.Marshal(newMessageBody)
+	if err != nil {
+		qmlLogin.LoginDataIsValid(false)
+		loger.Log.Warningf("Can`t marshal login message. %s", err)
+		qmlStatus.SendStatus("Login data can't be sent")
+		return
+	}
+	newMessage := messageService.Message{
+		Header: newMessageHeader,
+		Body:   newRawMessageBody,
+	}
+	marshaledMessage, err := messageService.MarshalMessage(&newMessage)
+	if err := connection.WriteMessage(websocket.TextMessage, marshaledMessage); err != nil {
+		qmlLogin.LoginDataIsValid(false)
+		loger.Log.Warningf("Can not send login data. %s", err)
+		qmlStatus.SendStatus("Login data can't be sent")
+		return
+	}
+	qmlStatus.SendStatus("Waiting for server response.")
+	loger.Log.Info("Login data sent to server. Waiting for response.")
+	//start listening for login response
+	//if all right - we need to start listening other channels
+	listener.AuthorizationChannel = make(chan messageService.Message)
```

Comment:
Need some timer for response, because we can wait for it for infinity amount of hours